src/black/trans.py

```diff
@@ -590,11 +590,21 @@ def make_naked(string: str, string_prefix: str) -> str:
             """
             assert_is_leaf_string(string)
             if "f" in string_prefix:
-                string = _toggle_fexpr_quotes(string, QUOTE)
-                # After quotes toggling, quotes in expressions won't be escaped
-                # because quotes can't be reused in f-strings. So we can simply
-                # let the escaping logic below run without knowing f-string
-                # expressions.
+                f_expressions = (
+                    string[span[0] : span[1]] for span in iter_fexpr_spans(string)
+                )
+                expressions_contain_printable_quotes = any(
+                    re.search(r".*[\'\"].*=", expression)
```

Comment:
This catches a little too much:

```
% black --line-length=20 --preview -c "log.info(f'{\"a\" == \"b\"} aaaaaa aa aa aa')"                  
log.info(
    f'{"a" == "b"} aaaaaa'
    " aa aa aa"
)
```

Here I'd expect the quotes to get reversed.

Reply:
Nice catch! Also `{ "a" == "b"= }` is valid, as is `{ "a" == "b" == "c" ... = }`. I'll try to find a better regex.

Reply:
And even `{ foo("bar",x=123) }` 🙃

Reply:
Like in https://github.com/psf/black/pull/4005/files#r1378259931 I'm fine with the regex as is — we can revisit with PEP 701 support. Although we should add some test cases for these.

I don't see a way to do it without negative lookbehind, something like `.*[\'\"].*(?<![=!:])=` maybe

Reply:
Yeah, currently looking at `.*[\'\"].*(?<![!:=])={1}(?!=)(?![^\s:])`

Reply:
And maybe it's better to be more consistent, even if the regex becomes a bit more complex